frontend/src/pages/projects/screens/spawner/useNotebookSizeState.ts

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import useNotebookDeploymentSize from '~/pages/projects/screens/detail/notebooks/useNotebookDeploymentSize';
+import { NotebookKind } from '~/k8sTypes';
+import { NotebookSize } from '~/types';
+import { getCustomNotebookSize } from '~/pages/projects/utils';
+import { useNotebookSize } from './useNotebookSize';
+
+export const useNotebookSizeState = (
+  existingNotebook: NotebookKind | undefined,
+): {
+  selectedSize: NotebookSize;
+  setSelectedSize: (size: string) => void;
+  sizes: NotebookSize[];
+} => {
+  const { size: notebookSize } = useNotebookDeploymentSize(existingNotebook);
+  const { sizes } = useNotebookSize();
+
+  const allSizes = React.useMemo(() => {
+    if (notebookSize) {
+      return sizes;
+    }
+    const CUSTOM_NOTEBOOK_SIZE = getCustomNotebookSize(existingNotebook);
+    if (existingNotebook && !sizes.some((size) => size.name === CUSTOM_NOTEBOOK_SIZE.name)) {
```

Comment:
`sizes` will never contain the `CUSTOM_NOTEBOOK_SIZE` entry. This logic is left over from a previous iteration.